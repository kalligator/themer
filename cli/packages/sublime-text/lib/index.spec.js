const { colors } = require('../../colors-default'),
  { render, renderInstructions } = require('./index'),
  plist = require('plist');

describe('themer Sublime Text theme generator', () => {
  const promisedFiles = Promise.all(render(colors, {}));

  it('should produce two files when given a color set with both a dark and a light theme', async () => {
    const files = await promisedFiles;
    expect(files.length).toBe(2);
    expect(files.some((file) => /dark/.test(file.name))).toBe(true);
    expect(files.some((file) => /light/.test(file.name))).toBe(true);
  });

  it('should produce valid plist files', async () => {
    const files = await promisedFiles;
    files.forEach((file) => {
      const contents = file.contents.toString('utf8');
      expect(plist.parse.bind(plist, contents)).not.toThrow();
    });
  });

  it('should provide installation instructions', async () => {
    const files = await Promise.all(render(colors));
    const instructions = renderInstructions(files.map(({ name }) => name));
    expect(instructions).toMatchSnapshot();
  });
});
