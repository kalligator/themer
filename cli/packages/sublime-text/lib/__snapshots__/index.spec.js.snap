// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`themer Sublime Text theme generator should provide installation instructions 1`] = `
"
1. Copy (or symlink) the generated theme files (\`themer-sublime-text-dark.tmTheme\` or \`themer-sublime-text-light.tmTheme\`) to the \`User/\` packages folder (you can see where this folder is located by choosing the \\"Browse Packages...\\" menu option in Sublime Text).
2. Choose the theme from the list of available color themes.
"
`;
